lib/srv/db/redis/connection.go

```diff
@@ -0,0 +1,75 @@
+/*
+
+ Copyright 2022 Gravitational, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+package redis
+
+import (
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// DefaultPort is Redis default port.
+const DefaultPort = "6379"
+
+// ConnectionOptions represent Redis connection options.
+type ConnectionOptions struct {
+	cluster bool
```

Comment:
I see that redis uses the term `cluster` in its connection parameters, so naming it the same way here makes sense. Maybe it would be worth a short comment saying this is "cluster mode" to help clarify for those of us not familiar with Redis connection parameters?

Reply:
@jimbishopp I agree. I should better explain what cluster mode is, but talking about cluster I think there is one more thing to discuss. Currently we want to support standalone and cluster Redis modes, but Redis has a few more. Another one that we can be interested in is Redis Sentinel. I'm not sure how many people would be interested in that, but changing now the URI parameter from `cluster=true` to `mode=[single,cluster,sentinel]` would make adding potencial new mode easier. Go-redis has already support for it https://redis.uptrace.dev/guide/sentinel.html#redis-server-client so sounds like "easy" thing to add.

@r0mant @smallinsky Any thoughts?

Reply:
I would prefer `mode` over a boolean even if we never added Sentinel.

Reply:
`mode` then. I'll change that.